### ISSUE TITLE
[Warrior] Rend doesn't proc Sword Specialization

### ISSUE THREAD

**Issue by vvranceanu** [chromiecraft/chromiecraft#7400]
### What client do you play on?

enGB

### Faction

Alliance

### Content Phase:

Generic

### Current Behaviour

[Rend](https://www.wowhead.com/wotlk/spell=6547/rend) doesn't proc  [Sword Specialization](https://www.wowhead.com/wotlk/spell=12815/sword-specialization)

### Expected Blizzlike Behaviour

It should

### Source

It is implied by existing implementation and spell description:

- [Rend](https://www.wowhead.com/wotlk/spell=6547/rend) already procs [Sudden Death](https://www.wowhead.com/wotlk/spell=29725/sudden-death)
- both [Sudden Death ](https://www.wowhead.com/wotlk/spell=29725/sudden-death)and [Sword Spec](https://www.wowhead.com/wotlk/spell=12815/sword-specialization) have the same description of hitting a target with a weapon, while [Rend](https://www.wowhead.com/wotlk/spell=6547/rend) says "Requires melee weapon"
- https://www.wowhead.com/wotlk/spell=12815/sword-specialization#comments :
![image](https://github.com/user-attachments/assets/68f75fff-25e6-4288-b432-dd7a43661cd5)

### Steps to reproduce the problem

0. .mod hp 2000000

1. Spam macro till full rage
/target Asdx
.cooldown
/targetlasttarget
/cast Bloodrage

2. Go to target dummy and spam till out of rage
/stopattack
/cast Sunder Armor
/stopattack

3. Go to Step 1

4. Check log/damage meter(some white hits will slip, but you can see on logs it wasnt a Sword Spec proc)

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/529e9140323edeef916bd6a1b6c97e13aa38db08

### Operating system

Ubuntu 22.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-BreakingNewsOverride](https://github.com/AnchyDev/BreakingNewsOverride)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [mod-zone-difficulty](https://github.com/azerothcore/mod-zone-difficulty)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-event-scripts](https://github.com/55Honey/Acore_eventScripts)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

**Comment by vvranceanu** [comment]
One more point to support this: Dr Damage addon shows extra damage for Sword Spec
![image](https://github.com/user-attachments/assets/bef04661-f75d-4a0d-a8e6-db78861f457b)

**Comment by Rorschach91** [comment]
I've checked all three reports and I can confirm them.
Thank you for the reports. If you find other issues feel free to report them.

**Comment by vvranceanu** [comment]
yeah, they should be taken all 3 into consideration at a time i think. the logic individually is hard to follow maybe. i also had to change the reasoning mid-writing the reports cause initially i thought rend procced sword spec, but then i swapped to mongoose when i confirmed it actually didn't. i might have messed up a bit with the logic as well.

but either way i think the commend and dr damage are solid proof for this one, while for the other 2 i was hoping a dev with core knowledge knows or can infer information.

also i think there should be 2 reports about sword spec and sudden death(if you consider sudden death issue valid as well)
